### DISCUSSION TITLE
inserting journal entry with ansible

### DISCUSSION THREAD

**Discussion by jcralbino** [netbox-community/ansible_modules#588] [Q&A] [answered]
I have tried to create a journal entry using the ansible modules but it seems we do not have this option?

Should this be created as a parameter for each object, where we can for example call netbox.vlan module and provide its journal entry information ? or as a separate module completely? 

Is someone using this feature already in a automated way?

**Comment by ryanmerolle** [comment, marked as answer]
This is not currently a feature.  Totally can be added as a feature request, or you can use the ansible_uri module in the interim to hit the netbox api target for journal entries.

**Comment by ollytom** [comment]
I've written an ansible module, `netbox_journal_entry` modeled after
the existing netbox_device.py, netbox_region.py etc. modules. The use
case at the place I work is:

1. upgrade the OS e.g. from Debian 11.4 to 11.5
2. write a journal entry for that device or VM that this specific upgrade completed

I could spend a bit of time tightening it up and make it reliably
idempotent. Then I could submit it as a PR and see what others think.
@jcralbino keep an eye out! Hope it helps :)

**Reply by ollytom** [reply]
Going through old code and found that this has now been implemented; see [netbox_journal_entry](https://galaxy.ansible.com/ui/repo/published/netbox/netbox/content/module/netbox_journal_entry/)
